src/textual_fastdatatable/backend.py

```diff
@@ -387,3 +414,206 @@ def py_str(_ctx: Any, arr: pa.Array) -> str | pa.Array | pa.ChunkedArray:
             arr = pc.call_function(udf_name, [arr])
         width: int = pc.max(pc.utf8_length(arr.fill_null("")).fill_null(0)).as_py()
         return width
+
+
+class PolarsBackend(DataTableBackend[pl.DataFrame]):
+
+    @classmethod
+    def from_file_path(
+        cls, path: Path, max_rows: int | None = None, has_header: bool = True
+    ) -> "PolarsBackend":
+        if path.suffix in [".arrow", ".feather"]:
+            tbl = pl.read_ipc(path)
+        elif path.suffix == ".arrows":
+            tbl = pl.read_ipc_stream(path)
+        elif path.suffix == ".json":
+            tbl = pl.read_json(path)
+        elif path.suffix == ".csv":
+            tbl = pl.read_csv(path, has_header=has_header)
+        else:
+            raise TypeError(f"Dont know how to load file type {path.suffix} for {path}")
+        return cls(tbl, max_rows=max_rows)
+
+    @classmethod
+    def from_pydict(
+        cls, pydict: dict[str, Sequence[str | int]], max_rows: int | None = None
+    ) -> "PolarsBackend":
+        return cls(pl.from_dict(pydict), max_rows=max_rows)
+
+    @classmethod
+    def from_dataframe(
+        cls, frame: pl.DataFrame, max_rows: int | None = None
+    ) -> "PolarsBackend":
+        return cls(frame, max_rows=max_rows)
+
+    def __init__(self, data: pl.DataFrame, max_rows: int | None = None) -> None:
```

Comment:
Would be good to throw a friendly exception here if Polars isn't installed (`pl is None`)